instrumentation/redisson-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/RedissonDbAttributesExtractor.java

```diff
@@ -11,34 +11,59 @@
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.Channel;
 import io.opentelemetry.instrumentation.api.db.RedisCommandSanitizer;
-import io.opentelemetry.instrumentation.api.tracer.DatabaseClientTracer;
-import io.opentelemetry.instrumentation.api.tracer.net.NetPeerAttributes;
-import io.opentelemetry.semconv.trace.attributes.SemanticAttributes.DbSystemValues;
+import io.opentelemetry.instrumentation.api.instrumenter.db.DbAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
 import java.net.InetSocketAddress;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
-import org.redisson.client.RedisConnection;
+import org.checkerframework.checker.nullness.qual.Nullable;
 import org.redisson.client.protocol.CommandData;
 import org.redisson.client.protocol.CommandsData;
 
-public class RedissonClientTracer
-    extends DatabaseClientTracer<RedisConnection, Object, List<String>> {
+final class RedissonDbAttributesExtractor extends DbAttributesExtractor<RedissonRequest> {
+
   private static final String UNKNOWN_COMMAND = "Redis Command";
 
-  private static final RedissonClientTracer TRACER = new RedissonClientTracer();
+  @Override
+  protected String system(RedissonRequest request) {
+    return SemanticAttributes.DbSystemValues.REDIS;
+  }
+
+  @Override
+  protected @Nullable String user(RedissonRequest request) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable String name(RedissonRequest request) {
+    return null;
+  }
 
-  private RedissonClientTracer() {
-    super(NetPeerAttributes.INSTANCE);
+  @Override
+  protected String connectionString(RedissonRequest request) {
+    Channel channel = request.getConnection().getChannel();
+    InetSocketAddress remoteAddress = (InetSocketAddress) channel.remoteAddress();
+    return remoteAddress.getHostString() + ":" + remoteAddress.getPort();
   }
 
-  public static RedissonClientTracer tracer() {
-    return TRACER;
+  @Override
+  protected String statement(RedissonRequest request) {
+    List<String> sanitizedStatements = sanitizeStatement(request);
+    switch (sanitizedStatements.size()) {
+      case 0:
+        return UNKNOWN_COMMAND;
+        // optimize for the most common case
+      case 1:
+        return sanitizedStatements.get(0);
+      default:
+        return String.join(";", sanitizedStatements);
+    }
   }
 
   @Override
-  protected String spanName(
-      RedisConnection connection, Object ignored, List<String> sanitizedStatements) {
+  protected String operation(RedissonRequest request) {
+    List<String> sanitizedStatements = sanitizeStatement(request);
```

Comment:
Do you know why we use the sanitized statements here instead of directly reading from `Command.getName()` or plural?

Reply:
I suppose it's a remnant of old tracer implementation - it should be fine to use `command.getName()` for operation.